pkg/fake/ec2api.go

```diff
@@ -140,7 +141,7 @@ func (e *EC2API) CreateFleetWithContext(_ context.Context, input *ec2.CreateFlee
 					e.CalledWithCreateLaunchTemplateInput.Add(lt)
 				}
 				instanceState := ec2.InstanceStateNameRunning
-				for i := 0; i < int(*input.TargetCapacitySpecification.TotalTargetCapacity); i++ {
+				for ; fulfilled < int(*input.TargetCapacitySpecification.TotalTargetCapacity); fulfilled++ {
```

Comment:
Why is this change needed?

Reply:
We overfulfill instances when multiple launch templates are involved. Adding more launch templates made it a little worse, so noticed it in some of the tests related to this change.

Reply:
> We overfulfill instances when multiple launch templates are involved

Maybe we can discuss offline but I'm not quite sure what you mean here. You're saying that our mocked API will try to create an instance for each LT that's passed through and this causes us to mock more instances in the API than we actually need?

Reply:
correct